polling/assets.go

```diff
@@ -298,6 +303,65 @@ func (d *DataSources) PullAllPEGAssets(oprversion uint8) (pa PegAssets, err erro
 	return pa, nil
 }
 
+// PullBestPrice pulls the best asset price we can find for a given asset.
+// Params:
+//		asset		Asset to pull pricing data
+//		reference	Time reference to determine 'staleness' from
+//		sources		Map of datasources to pull the price quote from.
+func (d *DataSources) PullBestPrice(asset string, reference time.Time, sources map[string]IDataSource) (pa PegItem, err error) {
+	if sources == nil {
+		// If our data sources passed in are nil, then we don't need to do cache wrapping.
+		// We should always have sources passed in, aside from unit tests.
+		sources = d.DataSources
+	}
+
+	// All the given data sources for the asset
+	sourceList := d.AssetSources[asset]
+
+	var prices []PegItem
+
+	// Eval all datasources from the reference time
+	for _, source := range sourceList {
+		var price PegItem
+		price, err = sources[source].FetchPegPrice(asset)
+		if err != nil {
+			continue
+		}
+
+		if price.Value != 0 {
+			prices = append(prices, price)
+
+			// We can break out if this is a 'good' price
+			//	Is it stale AND the market is open? We can expect stale quotes in a closed market
+			if reference.Sub(price.When) > d.staleDuration && IsMarketOpen(asset, reference) {
+				// This price quote is stale, keep fetching prices
+				continue
+			}
+
+			// This price is acceptable
+			break
+		}
+	}
+
+	// No prices found, and the last data source had an error
+	if err != nil && len(prices) == 0 {
+		return
+	}
+
+	// pa.When is uninitialized, so the reference.Sub(pa.When) == a very large value
+	mostRecent := reference.Sub(pa.When)
```

Comment:
this is an unnecessarily confusing way of doing it, you don't need 'mostrecent' at all, you can just do:

```go
	for _, price := range prices {
		if price.When > pa.When {
			pa = price
		}
	}
```

Reply:
Good point :+1: